mailclient.py

```diff
@@ -43,14 +63,70 @@
     JVM_MAIL = 'fklubjvmloss@gmail.com'
 elif MODE == "prod":
     JVM_MAIL = 'fklubjvmlosstest@gmail.com'
+    if not MAINTAINER_MAILS and not BACKUP_MAINTAINER_MAILS:
+        print(f"[ERROR] No emails are set for mode: {MODE}. Quitting", file=sys.stderr)
+        exit(1)
 else:
     exit(1)  # Exit early, as unknown mode
 
 print(f"Running JVM-loss in mode: {MODE}")
 
 
-# Sets up connection to gmail
+class Session:
+    """
+    Encapsulates behaviour of a running session of JVM-loss.
+    """
+
+    def __init__(self):
+        self.gmail_conn = setup_gmail_connection()
+        self.db_conn = database_setup()
+
+    def check(self):
+        print("Checking for new mails from JVM")
+        mails_read = check_for_mails(self.gmail_conn, self.db_conn)
+        print(f"Done checking for new mails from JVM, mails read: {mails_read}")
+
+        print("Checking for new queries")
+        mails_read = check_queries(self.gmail_conn, self.db_conn)
+        print(f"Done checking for new queries, queries processed: {mails_read}")
+
+
+def database_setup():
+    """
+    Loads a database from disk if exists, otherwise creates a clean one
+    :return: a db_conn to sqlite
+    """
+    # Check whether to resume using db or create a new one
+    if os.path.exists(DB_FILE_NAME):
+        db_conn = sqlite3.connect(DB_FILE_NAME)
+        # Get last event for status message
+        last_event = get_last_event(db_conn)
+        # Check that last event is not empty
+        if len(last_event) > 0:
+            print(f"UTC timestamp of last event in db: "
+                  f"{datetime.datetime.utcfromtimestamp(last_event[0][0]).strftime('%Y-%m-%d %H:%M:%S')}. "
+                  f"Total number of events: {len(get_events(db_conn))}")
+        return db_conn
+    else:
+        print("Database does not exist on disk, creating a new one")
+        return setup_clean_database()
+
+
+def setup_clean_database():
+    """
+    Creates a clean database.
+    FIXME: why is an ingredient event inserted into a clean db? should be elaborated here
```

Comment:
Because a dispenseddrinkevent is handled by subtracting the amount of ingredients from the last ingredient status in DB that the dispenseddrink used, and inserting these new values to DB. Would throw an error if there were no status in DB already. however, I am not sure if the first status should be empty, or full

Reply:
There are two approaches to assuming initial state;
- Assume empty, but then what happens when the machine allows for a dispense-event? Do we underflow or just omit updating jvm-loss tracked ingredient levels?
- Assume full, but then there's the issue with no notifications from the machine that a dispense event failed. In this case we must rely on 'ingredient below threshold' then.

I think we should just assume full ingredients when starting from a clean database.